spec/rspec/core/configuration_spec.rb

```diff
@@ -1819,6 +1819,20 @@ def exclude?(line)
         expect(group.metadata).to include(:b1_desc => "bar (block 1)", :b2_desc => "bar (block 1) (block 2)")
       end
 
+      it 'supports cascades of derived metadata, but avoids re-running derived metadata blocks that have already been applied' do
+        RSpec.configure do |c|
+          c.define_derived_metadata(:foo1) { |m| m[:foo2] = (m[:foo2] || 0) + 1 }
+          c.define_derived_metadata(:foo2) { |m| m[:foo3] = (m[:foo3] || 0) + 1 }
+          c.define_derived_metadata(:foo3) { |m| m[:foo4] = (m[:foo4] || 0) + 1 }
+        end
+
+        group = RSpec.describe("bar", foo1: 1)
+        expect(group.metadata).to include(foo1: 1, foo2: 1, foo3: 1, foo4: 1)
+
+        ex = RSpec.describe("My group").example("foo", foo1: 1)
+        expect(ex.metadata).to include(foo1: 1, foo2: 1, foo3: 1, foo4: 1)
```

Comment:
These are being treated as hashes, of course, but we still support 1.8.7, so we need the hash rocket syntax :)

Reply:
Thanks.  Fixed.